### PR TITLE
bug: fix Conn.Next not safe (#648)

### PR DESCRIPTION
**Pull request by zhongweikang**
## 1. Are you opening this pull request for bug-fixes, optimizations or new feature?
bug-fixes

## 2. Please describe how these code changes achieve your intention.

The code was modified at the suggestion of the author。

## 3. Please link to the relevant issues (if any).

https://github.com/panjf2000/gnet/issues/648

## 4. Which documentation changes (if any) need to be made/updated because of this PR?

nothing

## 4. Checklist

- [x] I have squashed all insignificant commits.
- [x] I have commented my code for explaining package types, values, functions, and non-obvious lines.
- [ ] I have written unit tests and verified that all tests passes (if needed).
- [ ] I have documented feature info on the README (only when this PR is adding a new feature).
- [x] (optional) I am willing to help maintain this change if there are issues with it later.

### CODE REVIEW

File: connection_unix.go [hunk 1]
```diff
@@ -325,10 +325,11 @@ func (c *conn) Next(n int) (buf []byte, err error) {
 	}
 	head, tail := c.inboundBuffer.Peek(n)
 	defer c.inboundBuffer.Discard(n) //nolint:errcheck
+	c.loop.cache.Reset()
 	if len(head) >= n {
-		return head[:n], err
+		c.loop.cache.Write(head)
```
Comment (panjf2000) [inline connection_unix.go:330]: 这一行不用放到这个 block 里，直接放外面就行了。

File: connection_windows.go [hunk 2]
```diff
@@ -164,10 +164,11 @@ func (c *conn) Next(n int) (buf []byte, err error) {
 	}
 	head, tail := c.inboundBuffer.Peek(n)
 	defer c.inboundBuffer.Discard(n) //nolint:errcheck
+	c.loop.cache.Reset()
 	if len(head) >= n {
-		return head[:n], err
+		c.loop.cache.Write(head)
```
Comment (panjf2000) [inline connection_windows.go:169]: 同上

File: connection_unix.go (hunk 1 above)
Comment (zhongweikang) [reply connection_unix.go]: 已改

File: connection_windows.go (hunk 2 above)
Comment (zhongweikang) [reply connection_windows.go]: 已改

**Comment by panjf2000** [comment]
Thanks!

**Pull request merged**